### DISCUSSION TITLE
Maui performance questions...

### DISCUSSION THREAD

**Discussion by King-Taz** [dotnet/maui#26117] [General]
Some under-the-hood questions regarding Maui performance...

Does Maui actually draw content with lower ZIndex?

Does Maui draw content outside the window (e.g., content in a ScrollView content)?

My (subjective) testing indicates Maui does consume compute cycles doing this. If that is correct, is there any way to tell Maui to stop doing that!

**Comment by stephenquan** [comment]
If you don't want off-screen components to be instantiated, consider using CollectionView. Even better, some third-party libraries such as SyncFusion or Telerik have even more optimized CollectionView variants in their library.

**Reply by King-Taz** [reply]
Thanks, CollectionView is a great idea (that I didn't even consider). I'm still curious though if compute time is spent on on-screen content with lower ZIndex and, if so, can this be eliminated. I don't find any evidence to suggest CollectionView works in this manner, for on-screen content that is over-lapped.

*FYI* It appears that ListView is also virtualized like CollectionView and only draws visible content. ListView seems primarily intended for single column content.

**Reply by King-Taz** [reply]
CollectionView is not a static control and is not useful in my app

**Comment by stephenquan** [comment]
If you have specific requirements, you can build the logic yourself. Define and track the geometry while implementing your own approach to distinguishing on-screen from off-screen.

The parent control can be a simple [AbsoluteLayout](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/layouts/absolutelayout?view=net-maui-9.0) or [Grid](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/layouts/grid?view=net-maui-9.0). With MVVM, you can manage children using [BindableLayout](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/layouts/bindablelayout?view=net-maui-9.0).

**Reply by King-Taz** [reply]
True, but my testing is subjective at this point. I'm hoping a MSFT developer would answer this definitively, regarding static controls. I suppose on mobile smaller mobile platforms redraw might not be an issue as you can only see so much information on a handheld device. But it's essential to know the software's limitations on larger mobile platforms (iPads, tablets, notebooks) and desktop apps that can show more information (not everyone is running i9 Intel with 40/50 series Nvidia).
